lib/Sema/CSFix.cpp

```diff
@@ -732,10 +732,34 @@ AllowFunctionTypeMismatch::create(ConstraintSystem &cs, Type lhs, Type rhs,
       AllowFunctionTypeMismatch(cs, lhs, rhs, locator, index);
 }
 
+bool GenericArgumentsMismatch::coalesceAndDiagnose(
+    const Solution &solution, ArrayRef<ConstraintFix *> secondaryFixes,
+    bool asNote) const {
+  std::set<unsigned> scratch(getMismatches().begin(), getMismatches().end());
+
+  for (auto *fix : secondaryFixes) {
+    auto *genericArgsFix = fix->castTo<GenericArgumentsMismatch>();
+    for (auto mismatchIdx : genericArgsFix->getMismatches())
+      scratch.insert(mismatchIdx);
+  }
+
+  SmallVector<unsigned> mismatches(scratch.begin(), scratch.end());
+  // Make sure that notes about generic arguments are always produced in order.
+  llvm::array_pod_sort(mismatches.begin(), mismatches.end());
```

Comment:
Not that it matters much in this case, but do we need to sort? `std::set` is a sorted container on `Compare` which defaults is `class Compare =std::less<Key>`, so `mismatches` would be constructed sorted already since is begin to end.

Reply:
That’s what I read as well but for some reason one of the tests produced notes out of order and I decided to just sort always to make sure there is no flakiness.

Reply:
Maybe there was something else going on while I was testing though, will give it a try without sort a couple of times to make sure…

Reply:
> Maybe there was something else going on while I was testing though, will give it a try without sort a couple of times to make sure…

No worries, I was just curious =]
Even looked for a bit in the `SmallVector` [implementation](https://llvm.org/doxygen/SmallVector_8h_source.html#l01215) to see if the order may be somehow lost there but it seems like not.

Reply:
Let's see if we can reproduce this, I might have been using `SmallSet` when I ran the tests...

Reply:
Looks like it’s fine :)